WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m

```diff
@@ -121,9 +125,27 @@ - (NSData *)fb_screenshotWithError:(NSError*__autoreleasing*)error
 #endif
 }
 
-- (NSData *)fb_rawScreenshotWithQuality:(NSUInteger)quality rect:(CGRect)rect error:(NSError*__autoreleasing*)error
+- (NSData *)fb_rawScreenshotWithQuality:(NSUInteger)quality error: (NSError*__autoreleasing*) error
 {
-  return [XCUIScreen.mainScreen screenshotDataForQuality:quality rect:rect error:error];
+  id<XCTestManager_ManagerInterface> proxy = [FBXCTestDaemonsProxy testRunnerProxy];
+  __block NSData *screenshotData = nil;
+  dispatch_semaphore_t sem = dispatch_semaphore_create(0);
+  void (^completion)(NSData *, NSError *) = ^(NSData *data, NSError *screenshotError) {
+    screenshotData = data;
+    *error = screenshotError;
+    dispatch_semaphore_signal(sem);
+  };
+  if ([self fb_isNewScreenshotAPISupported]) {
```

Comment:
cannot we simply keep the previous `XCUIScreen.mainScreen screenshotDataForQuality` method call here? Otherwise too many changes are needed

Reply:
Which change do you mean?